runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientFetchFactory.java

```diff
@@ -2066,6 +2070,10 @@ private final class KafkaFetchClient extends KafkaSaslClient
             private long nextOffset;
             private long stableOffset;
             private long latestOffset;
+            private long initialLatestOffset;
+            private long initialStableOffset;
+            private boolean dataFrameSent = false;
```

Comment:
These should be tracked on the `KafkaFetchStream`, not the `KafkaFetchClient`.
The `KafkaFetchStream` would then determine if anything needs to be done when the `FetchResponse` is decoded.

Also, rather than a boolean flag consider tracking `flushOrDataFramesSent` and incrementing in both `doApplicationData` and `doApplicationFlush`, then adding `doApplicationFlushIfNecessary` that checks if `flushOrDataFramesSent` is `0`.